### DISCUSSION TITLE
Need help to optimize edge requests & fast origin transfer

### DISCUSSION THREAD

**Discussion by Ariette** [vercel/community#7393] [Help]
### Summary

In April, I received an email notification regarding your updated pricing policy.
While my current usage falls within the Hobby plan limits, the new policy introduces a free tier with usage limits that I am currently exceeding. (Especially, Edge requests and Fast origin transfer)
I have been actively working to optimize my service to comply with the new free tier usage limits within the provided grace period, according to this docs https://vercel.com/docs/pricing/networking

To optimize my usage, I have implemented the followings:
  - Migrated most frequently accessed images from Vercel to a different server
  - Implemented a service worker to enforce browser caching of images
  - Disabled the prefetch feature of Next.js's Link component
  - Extended CDN cache duration for SSR-rendered pages to reduce execution frequency

I thought these will reduce the usage dramatically, but I have not achieved the expected optimization results.
In fact, it made almost zero effect.

[Edge requests]
![image](https://github.com/vercel/community/assets/10518158/d47bca32-25c3-4867-87bc-423cc66b9e9a)

[Fast origin transfer]
![image](https://github.com/vercel/community/assets/10518158/38e798e5-9b27-4e8f-97c7-f6c243d4870e)

According to GA on the website, the number of actual user is not increased. Then, how can it be?

I am concerned that I may be misunderstanding the new metrics or that my approach to optimization is flawed.
My website is a fan site that is entirely volunteer-run. Maintaining the Hobby plan is very important for me to continue operating the site.
Please let me know why it never decreases though the processes, and give me more information about optimizing the usage.
Thank you.

### Steps to Reproduce

N/A

**Comment by leerob** [comment]
- Could you please share the URL or more information about your site?
- What date did you deploy the changes? How long have you been monitoring?
- What is shown under "Top Paths" for requests?
- Where is your database located and how are you accessing it? Are you using Middleware or ISR?

The steps you've outlined to decrease usage seem correct so I don't believe those are wrong to do.

**Reply by Ariette** [reply]
> What date did you deploy the changes? How long have you been monitoring?

It was May 10th, and I've monitored about 2 months

> What is shown under "Top Paths" for requests?

Opps, I didn't know there is a 'Request' tab in 'Top Paths' pannel.
I checked the 'Top Paths' and found that `sw.js` and `workbox.js` are the most frequently called files.
Furthermore,  some static assets, such as `bg.png`, are being requested dozens of times more frequently than before, which is completely the opposite of what I had expected.
It is likely due to the precache configuration in the service worker.
I may have made a critical mistake when adding the service worker for asset caching.
Thank you for notice me about the 'Top Paths'

> Where is your database located and how are you accessing it? Are you using Middleware or ISR?

Data, thousands of static json file, are stored in github repo and distributed via github cdn(raw.githubusercontent.com)
I'm using SSR(via `getServerSideProps`) to reduce build time, because it takes too long time to build more than 30k pages everytime.

**Reply by leerob** [reply]
1. You might consider moving those static assets to [Vercel Blob storage](https://vercel.com/docs/storage/vercel-blob)
2. To decrease your Fast Origin Transfer, [add caching headers to the response](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props#caching-with-server-side-rendering-ssr) in `getServerSideProps`.

[a comment by an automated account is not shown]
